src/sys/linux/random.cr

```diff
@@ -30,6 +30,21 @@ module Sys
       end
     end
 
+    # Returns a random UInt32.
+    def self.random_number : UInt32
```

Comment:
I don't think you understand that pull request.
This whole `rand_type` business is already in the code base and already ensures that as few `next_u` invocations are done as needed, it's not tied to the integer type but to the size of the requested range. Copying it here (without attribution, too) is not changing anything.
The thing that makes a difference is the granularity of `next_u` which could return one byte instead of arbitrarily grabbing 4.
But then you start thinking about doing too many system calls, which my PR tries to mitigate with that `random_bytes` thing which is introduced as an alternative to `next_u`